cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedTable.java

```diff
@@ -180,6 +174,33 @@ public Scanner readByIndex(byte[] column, byte[] value) {
   }
 
   /**
+   * Reads table rows within the given secondary index key range. If no rows are indexed, falling within the given
+   * range, then a {@link co.cask.cdap.api.dataset.table.Scanner} with no results will be returned.
+   *
+   * @param column the column to use for the index lookup
+   * @param startValue the inclusive start of the range for which rows must fall within to be returned in the scan
+   * @param endValue the exclusive end of the range for which rows must fall within to be returned in the scan
+   * @return a Scanner returning rows from the data table, whose stored value for the given column is within the the
+   * given range.
+   * @throws java.lang.IllegalArgumentException if the given column is not configured for indexing.
+   */
+  public Scanner scanByIndex(byte[] column, byte[] startValue, byte[] endValue) {
```

Comment:
When doing range scans like this, you will often want to pass a null for start value or end value, indicating "all values from the beginning" and "all values from this start to the end".

I think supporting nullable values for these two parameters would make for a much nicer API.